lib/chef/resource/apt_package.rb

```diff
@@ -75,6 +90,10 @@ class AptPackage < Chef::Resource::Package
         description: "A Hash of response file variables in the form of {'VARIABLE' => 'VALUE'}.",
         default: {}, desired_state: false
 
+      property :anchor_package_regex, [TrueClass, FalseClass],
+        introduced: "18.3",
+        description: "A Boolean flag that allows (`false`) or prevents (`true`) apt_package from matching the named package with packages by regular expression if it can't find a package with the exact same name.",
```

Comment:
```suggestion
        description: "A Boolean flag that wraps a package name in regex anchors to help prevent `apt_package` from installing packages with similar names.",
```

Reply:
How does this prevent this? What do true and false do?

Reply:
Because the regex anchors specify that the regex pattern must include the entire package name.

Reply:
The larger description above provides lots of other context. But anything treated like a regular expression means it'll match substrings, so "apache" which would match "apache-mycoolthing" or whatever. This is the short-description, suggestions welcome

Reply:
This PR was merged with this change as well, and the description is not accurate as it stands. It claims it's not a regex anymore, but it still is, it's just an anchored one. If someone passes in foo.*bar that'll still work.